### PR TITLE
HPCC-30889 add Windows Clienttools 32bit build

### PR DESCRIPTION
**Pull request by xwang2713**
Add windows-2033 matrix for Clienttools 32bit
The only difference compared with x64 is "-T host=x86 -A Win32" in CMake configuration

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by xwang2713** [comment]
@Michael-Gardner can you help to run the github action again?  I switch host from x86x to x64. I think it means Windows OS not C/C++ compiler which is set by "-A Win32"

File: .github/workflows/build-assets.yml [hunk 1]
```diff
@@ -372,6 +372,11 @@ jobs:
     strategy:
       matrix:
         include:
+          - os: "windows-2022"
+            mono: ""
+            sudo: ""
+            cmake_config_options: '-T host=x64 -A Win32 -DUSE_OPTIONAL=OFF -DCLIENTTOOLS_ONLY=ON -DUSE_AZURE=OFF -DUSE_CASSANDRA=OFF -DUSE_JAVA=OFF -DUSE_OPENLDAP=OFF'
```
Comment (GordonSmith) [inline .github/workflows/build-assets.yml:378]: Generally I don't specify the `-T host=x64` and let it use the default (which may well be the x64 toolchain)`.  I suspect this is 100% fine, just never tried it...

**Comment by GordonSmith** [comment]
@ghalliday - trying to remember, but did we officially drop support for Win32 client tools?

**Review by GordonSmith — APPROVED** [review]
Change looks ok - I would be inclined to remove the "-T host=" part.
I do wonder if we want to support Win32 from 9.6.x onwards?

**Comment by ghalliday** [comment]
@GordonSmith I also cannot remember.
The jira suggests that it is needed for the ecl ide build.  Is that still the case, and is it the case for 8.12.x?
I would expect this PR to target 8.12.x, but for us to then remove support for all builds where it is no longer required.

**Pull request merged**